### PR TITLE
[SHIP-3004] ZKSync: implement uptime feed validator

### PR DESCRIPTION
**Pull request by jlaveracll**
### Resolves Dependencies

TODO:
- Fix the `matter-labs/contracts` repo branch/tag/version
- Use [`zks_getbridgehubcontract`](https://docs.zksync.io/build/api-reference/zks-rpc#zks_getbridgehubcontract) to get the contract address?
- Add zksync contracts to `native_solc_compile_all_l2ep`?
- Add `native_solc_compile_all_l2ep` to `native_solc_compile_all`?

### CODE REVIEW

File: contracts/src/v0.8/l2ep/dev/zksync/ZKSyncValidator.sol [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    bytes4 selector = ZKSyncSequencerUptimeFeedInterface.updateStatus.selector;
+    bool status = currentAnswer == ANSWER_SEQ_OFFLINE;
+    uint64 timestamp = uint64(block.timestamp);
+    // Encode `status` and `timestamp`
+    bytes memory message = abi.encodeWithSelector(selector, status, timestamp);
+    bytes[] memory emptyBytes;
+
+    // Create the L2 transaction request
+    L2TransactionRequestDirect memory l2TransactionRequestDirect = L2TransactionRequestDirect({
+      chainId: 1, // TODO look at docs
```
Comment (mohamed-mehany) [inline contracts/src/v0.8/l2ep/dev/zksync/ZKSyncValidator.sol]: I think:
300 for testnet
324 for mainnet.

File: contracts/src/v0.8/l2ep/dev/zksync/ZKSyncValidator.sol [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      mintValue: 0,
+      l2Contract: L2_UPTIME_FEED_ADDR,
+      l2Value: 0, // TODO confirm
```
Comment (mohamed-mehany) [inline contracts/src/v0.8/l2ep/dev/zksync/ZKSyncValidator.sol]: I think we need to use this function to estimate the value: https://github.com/matter-labs/era-contracts/blob/7ca5517510f2534a2fc25b16c429fdd4a439b89d/l1-contracts/contracts/bridgehub/Bridgehub.sol#L202

File: contracts/src/v0.8/l2ep/dev/zksync/ZKSyncValidator.sol [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      l2Calldata: message,
+      l2GasLimit: s_gasLimit,
+      l2GasPerPubdataByteLimit: 0, // TODO confirm
```
Comment (mohamed-mehany) [inline contracts/src/v0.8/l2ep/dev/zksync/ZKSyncValidator.sol]: The zksync team mentioned that a static value can be used for this.
So maybe we should make it configurable via args similar to the gas limit.

File: contracts/src/v0.8/l2ep/dev/zksync/ZKSyncValidator.sol (hunk 1 above)
Comment (jlaveracll) [reply contracts/src/v0.8/l2ep/dev/zksync/ZKSyncValidator.sol]: got it! do you have a source for those values so I can link some docs?

Comment (mohamed-mehany) [reply contracts/src/v0.8/l2ep/dev/zksync/ZKSyncValidator.sol]: I found this one: https://docs.zksync.io/build/connect-to-zksync

**Pull request closed without merging**